lib/internal/streams/compose.js

```diff
@@ -94,15 +98,53 @@ module.exports = function compose(...streams) {
   });
 
   if (writable) {
-    d._write = function(chunk, encoding, callback) {
+    writableEndDestructor = makeWritableEnd(d, head, tail);
+  }
+
+  if (readable) {
+    readableEndDestructor = makeReadableEnd(d, head, tail);
+  }
+
+  d._destroy = function(err, callback) {
+    if (!err && onclose !== null) {
+      err = new AbortError();
+    }
+
+    if (readableEndDestructor) {
+      readableEndDestructor();
+    }
+
+    if (writableEndDestructor) {
+      writableEndDestructor();
+    }
+
+    if (onclose === null) {
+      callback(err);
+    } else {
+      onclose = callback;
+      if (isNodeStream(tail)) {
+        destroyer(tail, err);
```

Comment:
Some help is needed here, how could we destroy webstreams here?

Reply:
Can you elaborate on the question? (We can destroy web streams the question is what scenario do you specifically mean)

Reply:
When the pipeline encounters an error, it would call d.destroy, which in turn would destroy the last stream in the series the `tail` stream should the same happen for webstreams too I think we could do `writableStream.abort()` here.

Actually, i am a little confused why destroying the last stream is necessary 😅

Reply:
This one question remain wdyt @benjamingr ?